src/server.py

```diff
@@ -0,0 +1,168 @@
+# encoding: utf-8
+"""Small localtcp/ip server to host connections from local client."""
+
+import socket
+import datetime
+import os
+from mimetypes import MimeTypes
+
+walk_dir = os.getcwd()
+if walk_dir.split('/')[-1] != 'src':  # code to make tox work
+    walk_dir += '/src'
+walk_dir += '/webroot'
+
+
+buffer_length = 1024
+address = ('127.0.0.1', 5000)
+
+responses = {
+    400: ('Bad Request', 'Bad request syntax or unsupported method'),
+    404: ('Not Found', 'Nothing matches the given URI'),
+    405: ('Method Not Allowed', 'Specified method is invalid for this resource.'),
+    500: ('Internal Server Error', 'Server got itself in trouble'),
+    505: ('HTTP Version Not Supported', 'Cannot fulfill request.')
+}
+
+
+def response_ok(body, content_type):
+    """Return a well formed HTTP 200 OK response."""
+    response = b'HTTP/1.1 200 OK\r\n'
+    response += b'Date: ' + str(datetime.datetime.now().strftime('%a %b %Y %X PST')).encode() + b'\r\n'
+    response += b'Connection: close\r\n'
+    response += b'Content-Type: ' + content_type + b'\r\n'
+    response += b'Content-Length: ' + str(len(body)).encode() + b'\r\n\r\n'
+    response += body + b'\r\n'
```

Comment:
Instead of building this piece by piece, why not just make one large string and leave a placeholder for the response body that you can fill with string formatting?